source/gloo_proxy.go

```diff
@@ -93,6 +117,32 @@ func NewGlooSource(dynamicKubeClient dynamic.Interface, kubeClient kubernetes.In
 	}, nil
 }
 
+// virtualHosts aggregates all virtual hosts from the available listener types.
+// Only 1 of HybridListener or HTTPListener will be set
+func (l proxySpecListener) virtualHosts() []proxyVirtualHost {
+	virtualHosts := l.HTTPListener.VirtualHosts
+	for _, hybridMatchedListener := range l.HybridListener.MatchedListeners {
```

Comment:
do we have guarantees that both `l.HTTPListener` and `l.HybridListener` will not be nil before dereferencing both?

Reply:
nevermind, i see these are structs not pointers to structs